guides/micronaut-kafka/books/java/src/main/java/example/micronaut/AnalyticsFilter.java

```diff
@@ -0,0 +1,33 @@
+package example.micronaut;
+
+import io.micronaut.http.HttpRequest;
+import io.micronaut.http.MutableHttpResponse;
+import io.micronaut.http.annotation.Filter;
+import io.micronaut.http.filter.HttpServerFilter;
+import io.micronaut.http.filter.ServerFilterChain;
+import io.reactivex.Flowable;
+import org.reactivestreams.Publisher;
+
+import java.util.Optional;
+
+@Filter("/books/?*") // <1>
+public class AnalyticsFilter implements HttpServerFilter { // <2>
+
+    private final AnalyticsClient analyticsClient;
+
+    public AnalyticsFilter(AnalyticsClient analyticsClient) { // <3>
+        this.analyticsClient = analyticsClient;
+    }
+
+    @Override
+    public Publisher<MutableHttpResponse<?>> doFilter(HttpRequest<?> request,
+                                                      ServerFilterChain chain) { // <4>
+        return Flowable
+                .fromPublisher(chain.proceed(request)) // <5>
+                .switchMap(response -> {
+                    Optional<Book> book = response.getBody(Book.class); // <6>
+                    book.ifPresent(analyticsClient::updateAnalytics); // <7>
```

Comment:
The `updateAnalytics` method here is synchronous and blocking. It should probably return a reactive return type (see https://micronaut-projects.github.io/micronaut-kafka/latest/guide/index.html#kafkaClientMethods) so that it uses the kafka asynchronous API and doesn't block the event loop

Reply:
@burtbeckwith can you please do the same change in the RabbitMQ guide, because the code is the same for both.

Reply:
I was using the code from the RabbitMQ guide but oversimplified it. I updated to use the original.